shipyard/src/main/java/org/jclouds/shipyard/features/ContainersApi.java

```diff
@@ -64,6 +66,7 @@
    void restartContainer(@PathParam("id") String id);
    
    @Named("containers:deploy")
+   @Fallback(NullOnContainerResourceUnavailableAnd500.class)
```

Comment:
This is a funny one and an open bug for Shipyard. The deployContainer call only deploys a single container but returns a JSON array which is why we return the list here. There will only ever be 1 or 0 elements.

With that said yes an ResourceNotFoundExeption makes more sense.